test/Router.t.sol

```diff
@@ -156,6 +161,62 @@ contract RouterTest is Test, LogicSignature, DelegationSignature {
         if (agent != executingAgent) revert();
     }
 
+    function testExecuteBySig() external {
+        vm.prank(user);
+        router.newAgent();
+        IParam.Logic[] memory logics = new IParam.Logic[](1);
+        logics[0] = IParam.Logic(
+            address(mockTo), // to
+            '',
+            inputsEmpty,
+            IParam.WrapMode.NONE,
+            address(0), // approveTo
+            address(0) // callback
+        );
+        // Ensure correct EIP-712 encodeData
+        uint256 deadline = block.timestamp + 3600;
+        uint256 nonce = 0;
+        IParam.ExecutionDetails memory details = IParam.ExecutionDetails(
+            logics,
+            tokensReturnEmpty,
+            SIGNER_REFERRAL,
+            nonce,
+            deadline
+        );
+        bytes memory signature = getTypedDataSignature(details, router.domainSeparator(), userPrivateKey);
+        vm.expectEmit(true, true, true, true, address(router));
+        emit Execute(user, address(router.agents(user)), SIGNER_REFERRAL);
+        router.executeBySig(details, user, signature);
+        assertFalse(router.getAgent(user) == address(0));
```

Comment:
Why do we need to check the agent address if we prank the `user` to create a new agent in Ln.166?

Reply:
Indeed. Removed.